airflow/contrib/operators/pubsub_operator.py

```diff
@@ -91,29 +92,289 @@ def execute(self, context):
+            )
+
+    Finally, subscription is not required. If not passed, the operator will
+    generated a universally unique identifier for the subscription's name. ::
+
+        with DAG('DAG') as dag:
+            (
+                dag >> PubSubSubscriptionCreateOperator(
+                    topic_project='my-project', topic='my-topic')
+            )
+
+    ``topic_project``, ``topic``, ``subscription``, and
+    ``subscription`` are templated so you can use variables in them.
+    """
+    template_fields = ['topic_project', 'topic', 'subscription',
+                       'subscription_project']
+    ui_color = '#0273d4'
+
+    @apply_defaults
+    def __init__(
+            self,
+            topic_project,
+            topic,
+            subscription=None,
+            subscription_project=None,
+            ack_deadline_secs=10,
+            fail_if_exists=False,
+            gcp_conn_id='google_cloud_default',
+            delegate_to=None,
+            *args,
+            **kwargs):
+        """
+        :param topic_project: the GCP project ID where the topic exists
+        :type topic_project: string
+        :param topic: the topic to create. Do not include the
+            full topic path. In other words, instead of
+            ``projects/{project}/topics/{topic}``, provide only
+            ``{topic}``. (templated)
+        :type topic: string
+        :param subscription: the Pub/Sub subscription name. If empty, a random
+            name will be generated using the uuid module
+        :type subscription: string
+        :param subscription_project: the GCP project ID where the subscription
+            will be created. If empty, ``topic_project`` will be used.
+        :type subscription_project: string
+        :param ack_deadline_secs: Number of seconds that a subscriber has to
+            acknowledge each message pulled from the subscription
+        :type ack_deadline_secs: int
+        :param gcp_conn_id: The connection ID to use connecting to
+            Google Cloud Platform.
+        :type gcp_conn_id: string
+        :param delegate_to: The account to impersonate, if any.
+            For this to work, the service account making the request
+            must have domain-wide delegation enabled.
+        :type delegate_to: string
+        """
+        super(PubSubSubscriptionCreateOperator, self).__init__(*args, **kwargs)
+
+        self.topic_project = topic_project
+        self.topic = topic
+        self.subscription = subscription
+        self.subscription_project = subscription_project
+        self.ack_deadline_secs = ack_deadline_secs
+        self.fail_if_exists = fail_if_exists
+        self.gcp_conn_id = gcp_conn_id
+        self.delegate_to = delegate_to
+
+    def execute(self, context):
+        hook = PubSubHook(gcp_conn_id=self.gcp_conn_id,
+                          delegate_to=self.delegate_to)
+
+        hook.create_subscription(
```

Comment:
return hook.create_subscription(...) to make it auto push to XCOM, otherwise an anonymous subscription is not useful, fine with forcing non-empty subscription on the other hand.